### PR TITLE
Fix advanced search

### PR DESCRIPTION
**Pull request by Hartorn**
This PR is mainly about Advanced-Search. 
Must be treated with same PR on focus-core

The main problems are listeners not properly recreated when store or store properties are changing

see https://github.com/KleeGroup/focus-core/pull/394

### CODE REVIEW

**Comment by Hartorn** [comment]
Still need to correct 
* focus-components/common/form/mixin/reference-behaviour
* focus-components/common/mixin/store-change-behaviour
* focus-components/common/mixin/store-behaviour

**Comment by Hartorn** [comment]
@c3dr0x Would be great if you could do some test on this branch, using a webpack alias

**Comment by Hartorn** [comment]
@c3dr0x Just Rebased it

File: src/common/form/mixin/reference-behaviour.js [hunk 1]
```diff
@@ -21,43 +22,62 @@ const referenceMixin = {
...
+        const safeOldReferenceNames = oldReferenceNames || [];
+
+        if (this.addStoreSub && this.removeStoreSub) { 
+            const toAdd = difference(safeReferenceNames, safeOldReferenceNames);
+            const toDelete = difference(safeOldReferenceNames, safeReferenceNames);
 
-        //If the reference store is empty don't do anything.
-        if(isEmpty(this.referenceNames)) {
-            return;
+            if (this.addStoreSub && this.removeStoreSub) {
```
Comment (c3dr0x) [inline src/common/form/mixin/reference-behaviour.js]: Test for `addStoreSub` & `removeStoreSub` is already done 4 lines before

File: src/common/mixin/store-behaviour.js [hunk 2]
```diff
@@ -133,16 +133,68 @@ const storeMixin = {
...
-    /** @inheritdoc */
+    addStoreSub(store, property) {
+        if (!this.stores) {
+            this.stores = [];
+        }
+
+        const storeIndex = findIndex(this.stores, elt => elt.store === store);
+        const existingConf = storeIndex === -1 ? null : this.stores[storeIndex];
+
+        if (existingConf && existingConf.properties.indexOf(property) !== -1) {
```
Comment (c3dr0x) [inline src/common/mixin/store-behaviour.js]: testing with `includes` is better here

File: src/common/mixin/store-behaviour.js [hunk 3]
```diff
@@ -133,16 +133,68 @@ const storeMixin = {
...
+            return ;
+        }
+        if (!store || !store.definition || !store.definition[property]) {
```
Comment (c3dr0x) [inline src/common/mixin/store-behaviour.js]: Lines 147-152 should be refactored with content of `_registerListeners`

File: src/common/mixin/store-behaviour.js [hunk 4]
```diff
@@ -133,16 +133,68 @@ const storeMixin = {
...
+        const storeIndex = findIndex(this.stores, elt => elt.store === store);
+        if (storeIndex === -1) {
+            return ;
+        }
+        const existingConf = this.stores[storeIndex];
+        const propertyIndex = existingConf.properties.indexOf(property);
+        if (propertyIndex === -1) {
+            return ;
+        }
+        if (!store || !store.definition || !store.definition[property]) {
```
Comment (c3dr0x) [inline src/common/mixin/store-behaviour.js]: Lines 176-182 should be refactored with content of _unRegisterListeners

**Pull request merged**